sockeye/encoder.py

```diff
@@ -49,14 +56,30 @@ def get_encoder(num_embed: int,
     :param dropout: Dropout probability for encoders (RNN and embedding).
     :param forget_bias: Initial value of RNN forget biases.
     :param fused: Whether to use FusedRNNCell (CuDNN). Only works with GPU context.
+    :param char_seq_encoder: Whether to use CharacterSequenceEncoder after source embedding.
+    :param cse_max_filter_width: Maximum filter width for CharacterSequenceEncoder.
+    :param cse_num_filters: Number of filters of each width for CharacterSequenceEncoder.
+    :param cse_pool_stride: Pooling stride for CharacterSequenceEncoder.
+    :param cse_num_highway_layers: Number of highway layers for CharacterSequenceEncoder.
     :return: Encoder instance.
     """
     # TODO give more control on encoder architecture
     encoders = list()
+
     encoders.append(Embedding(num_embed=num_embed,
                               vocab_size=vocab_size,
                               prefix=C.SOURCE_EMBEDDING_PREFIX,
                               dropout=dropout))
+
+    if char_seq_encoder:
+        encoders.append(CharacterSequenceEncoder(num_embed=num_embed,
+                                                 max_filter_width=cse_max_filter_width,
+                                                 num_filters=cse_num_filters,
+                                                 pool_stride=cse_pool_stride,
+                                                 num_highway_layers=cse_num_highway_layers,
+                                                 prefix=C.CHAR_SEQ_ENCODER_PREFIX,
+                                                 dropout=dropout))
+
```

Comment:
Lets have multiple functions that define reasonable defaults for different types of encoders and switch on calling these functions based on the --encoder flag